### DISCUSSION TITLE
Throwing Route Error from the client

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [remix-run/remix#8124] [Q&A]
Hi,
I'm having a case where I'm loading some data on the client with fetch, and want to show a 404 page in the context of layout, but I need to throw it directly from the client. 

Is that possible in Remix? 

Thanks!

**Comment by sergiodxa** [comment]
If you throw any error it should render the ErrorBoundary. If it doesn't work from within the effect or event handler try to do it from the component itself.

**Reply by piotrkulpinski** [reply]
It works, but it's not a Route Error, hence I can't use `useRouteError` in the ErrorBoundary.
